src/WooCommerce.php

```diff
@@ -740,6 +740,52 @@ public static function woocommerce_save_product_variation($post_id, $loop) {
      update_post_meta($variation_id, self::FIELD_MARKETING_FOCUS, $marketingFocusFieldValue);
   }
 
+  /**
+   * Allows adding multiple products to the cart.
+   *
+   * See https://bit.ly/3JfcIET
+   *
+   * @implements wp_loaded
+   */
+  public static function add_multiple_products_to_cart() {
+    if (
+      !isset($_REQUEST['add-to-cart']) ||
+      strpos($_REQUEST['add-to-cart'], ',') === false
+    ) {
+      return;
+    }
+
+    remove_action('wp_loaded', array('WC_Form_Handler', 'add_to_cart_action'), 20);
+
+    $product_ids = explode(',', $_REQUEST['add-to-cart']);
+    if (!$product_ids) {
+      return;
+    }
+    $num_products = count($product_ids);
+
+    $adding_to_cart = TRUE;
+    $redirect_to_cart = get_option('woocommerce_cart_redirect_after_add');
+
+    // Avoid redirection to cart after adding a product.
+    $fn_return_no = fn() => 'no';
+    add_filter('woocommerce_add_to_cart_redirect', '__return_false', 99);
+    add_filter('option_woocommerce_cart_redirect_after_add', $fn_return_no, 99);
+
+    foreach ($product_ids as $index => $product_id) {
+      $product_id = apply_filters('woocommerce_add_to_cart_product_id', absint($product_id));
+      if (empty($product_id)) {
```

Comment:
I just found a similar implementation for AJAX calls at https://github.com/netzstrategen/wordpress-shop-styleboards/blob/c82b63b2a56dc57a2f4622a8e41caf38dcad2f7e/src/Plugin.php#L644-L679, I think we should validate first before adding to cart (have not tested this code).
```suggestion
      $passed_validation = apply_filters('woocommerce_add_to_cart_validation', TRUE, $product_id, 1);
      if ($passed_validation) {
```

Reply:
AFAIK hook `woocommerce_add_to_cart_validation` is used to allow custom validation when adding a product to the cart. I think is a good idea to use it, yes.